core/roboconf-messaging-rabbitmq/src/main/java/net/roboconf/messaging/rabbitmq/RabbitMqClientFactory.java

```diff
@@ -0,0 +1,177 @@
+	public synchronized void setMessageServerUsername( final String messageServerUsername ) {
+		this.messageServerUsername = messageServerUsername;
+		this.logger.finer("Server username set to " + messageServerUsername);
+	}
+
+	@Property(name = MessagingConstants.RABBITMQ_SERVER_PASSWORD, value = RabbitMqClient.DEFAULT_PASSWORD)
+	public synchronized void setMessageServerPassword( final String messageServerPassword ) {
+		this.messageServerPassword = messageServerPassword;
+		this.logger.finer("Server password set to " + messageServerPassword);
+	}
+
+	@Updated
+	public void reconfigure() {
+		// Set the properties for all created clients.
+		resetClients(false);
+	}
+
+	@Invalidate
+	public void stop() {
+		resetClients(true);
+	}
+
+	private void resetClients(boolean shutdown) {
+		// Make fresh snapshots of the clients, as we don't want to reconfigure them while holding the lock.
+		final ArrayList<RabbitMqClient> clients;
+		synchronized (this) {
+			clients = new ArrayList<>(this.clients);
+		}
+
+		// Now reconfigure all the clients.
+		for (RabbitMqClient client : clients) {
+			try {
+				final ReconfigurableClient<?> reconfigurable = client.getReconfigurableClient();
+				if (reconfigurable != null)
+					if (shutdown)
+						reconfigurable.closeConnection();
+					else
+						reconfigurable.switchMessagingType(MessagingConstants.FACTORY_RABBIT_MQ);
+			} catch (Throwable t) {
+				// Warn but continue to reconfigure the next clients!
+				this.logger.warning("A client has thrown an exception on reconfiguration: " + client);
+				Utils.logException(this.logger, new RuntimeException(t));
+			}
+		}
+	}
+
+	@Override
+	public String getType() {
+		return MessagingConstants.FACTORY_RABBIT_MQ;
+	}
+
+	@Override
+	public synchronized IDmClient createDmClient( final ReconfigurableClientDm parent ) {
+		final RabbitMqClientDm client = new RabbitMqClientDm(parent);
+		client.setParameters(this.messageServerIp, this.messageServerUsername, this.messageServerPassword);
+		clients.add(client);
+		this.logger.finer("Created a new DM client");
+		return client;
+	}
+
+	@Override
+	public synchronized IAgentClient createAgentClient( final ReconfigurableClientAgent parent ) {
+		final RabbitMqClientAgent client = new RabbitMqClientAgent(parent);
+		client.setParameters(this.messageServerIp, this.messageServerUsername, this.messageServerPassword);
+		clients.add(client);
+		this.logger.finer("Created a new Agent client");
+		return client;
+	}
+
+	@Override
+	public boolean setConfiguration( final Map<String, String> configuration ) {
+		final boolean result;
+		final String type = configuration.get(MESSAGING_TYPE_PROPERTY);
+		String ip = configuration.get(MessagingConstants.RABBITMQ_SERVER_IP);
+		String username = configuration.get(MessagingConstants.RABBITMQ_SERVER_USERNAME);
+		String password = configuration.get(MessagingConstants.RABBITMQ_SERVER_PASSWORD);
+		if (MessagingConstants.FACTORY_RABBIT_MQ.equals(type)) {
```

Comment:
Get configuration valies in the if block.
Question: do we have **org.apache.commons.collections** in our Maven dependencies? Because in this case, we could use **DefaultedMap**.